### DISCUSSION TITLE
Set default configuration

### DISCUSSION THREAD

**Discussion by plerup** [microsoft/vscode-cpptools#9912] [General]
I generate a c_cpp_properties.json by a separate program and then use this for a VS Code project.
Is there some way to select which of the configurations that is active on startup when you have several of them in your c_cpp_properties.json file? On startup the one selected in the previous run is the default and when the project is new the last one is selected. Can this be controlled in some way?

**Comment by sean-mcmanus** [comment]
It can be controlled via `vscode.commands.executeCommand('cpptools.setActiveConfigName', "configName")` from another extension, see https://github.com/microsoft/vscode-cpptools/issues/4870

**Comment by plerup** [comment]
Ok, thanks, got it working with https://marketplace.visualstudio.com/items?itemName=gabrielgrinberg.auto-run-command
